build-docker-images/action.yml

```diff
@@ -0,0 +1,33 @@
+name: Build Docker Images
+description: >
+  Builds multiple Docker images in parallel (with BuildKit enabled).
+  Each Docker image will be tagged with the name of its target.
+
+inputs:
+  path:
+    description: Docker build context path (e.g., ".").
+    required: true
+  targets:
+    description: Docker build targets as a space-separated list (e.g., "daemon server").
+    required: true
+  docker-build-extra-args:
+    description: Docker build extra arguments.
+    required: false
+
+runs:
+  using: composite
+  steps:
+    - shell: bash
+      run: |
+        for target in ${{ inputs.targets }}; do
+          DOCKER_BUILDKIT=1 \
+            docker build \
+              --build-arg COMMIT_SHA=${{ github.sha }} \
+              --ssh default \
+              --tag $target \
+              --target $target \
+              ${{ inputs.docker-build-extra-args }} \
+              . &
+        done
+
+        wait
```

Comment:
🧠 clever

Reply:
Thanks! Something something shoulders of giants.